### ISSUE TITLE
🐛 `NavigationAppBar`'s `automaticallyImplyLeading` behavior is wrong  

### ISSUE THREAD

**Issue by mateusfccp** [bdlukaa/fluent_ui#1051]
**Describe the bug**
Currently, `automaticallyImplyLeading` just puts a back arrow or remove it. Even when there is no route below the current one (example, if you use it in a root page), an back arrow will appear.

**To Reproduce**
Use an `NavigationAppBar` with `automaticallyImplyLeading: true`, which is the default. Note that the back arrow button will be the leading even if there's no route below.

**Expected behavior**
The back arrow should appears if, and only if, there's a route below, which the behavior of popping the current route.

The same behavior as Flutter's Material `AppBar`. It even uses the same documentation.

**Comment by bdlukaa** [comment]
In some Windows apps, the arrow is displayed even when there is no route to pop. This behavior is usually true when the app consider the navigation view as a route history.

![WinUI 3 Gallery](https://github.com/bdlukaa/fluent_ui/assets/45696119/cfbe6e09-63f3-4577-b2a5-4a8bb9ed486f)
_WinUI 3 Gallery_

On other apps, this behavior is not respected:

![Microsoft Store Home](https://github.com/bdlukaa/fluent_ui/assets/45696119/0badfa27-d6c3-4565-90e3-f9a023ad7977)
![Microsoft Store Games](https://github.com/bdlukaa/fluent_ui/assets/45696119/1a3206f6-bc21-45aa-8782-bd66525376a3)

_Microsoft Store_

We could, indeed, add an option to customize this behavior, but it is really hard to follow a pattern when there is not a pattern.

We follow the guidance provided in the [`NavigationView` documentation](https://learn.microsoft.com/en-us/windows/apps/design/controls/navigationview):
![](https://learn.microsoft.com/en-us/windows/apps/design/controls/images/navigation-view-left-style.png)

Any major changes to `NavigationView` should be made after #919 is addressed.

**Comment by mateusfccp** [comment]
You are right, I just read the documentation and there are few instructions about the back button.

<img width="875" alt="image" src="https://github.com/bdlukaa/fluent_ui/assets/4605213/70baf552-b4b3-4d71-9a31-39714a2c04ba">
<img width="878" alt="image" src="https://github.com/bdlukaa/fluent_ui/assets/4605213/d124d882-a1bb-45e6-8e72-d8f6b2dffd25">

Considering this documentation, I think `automaticallyImplyLeading` is a misleading name, specailly because it is the same name and uses the same documentation as the `Material` counterpart.

Instead, then, we should use the same name as in the documentation, that is, `isBackButtonVisible`.

**Comment by Matzu89** [comment]
Is there a way to tap into the backbutton pressed event? Like a optional callback onBackButtonPressed, so that it can be forwarded to the responsible navigation context?
